PatientWebApplication/HealthClinic.DAL/Service/RegularAppointmentService.cs

```diff
@@ -145,5 +153,62 @@ public Boolean isTermNotAvailable(DoctorUser doctor, TimeSpan time, String dateT
+
+            appointments = SearchForDate(appointments, appointmentReportSearchDto);
+
+            appointments = SearchForDoctorNameAndSurname(appointments, appointmentReportSearchDto);
+
+            appointments = SearchForAppointmentType(appointments, appointmentReportSearchDto);
+
+            return appointments;
+
+        }
+
+        private List<DoctorAppointment> SearchForDoctorNameAndSurname(List<DoctorAppointment> appointments, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (!appointmentSearchDto.DoctorNameAndSurname.Equals(""))
+            {
+                appointments = appointments.FindAll(appointment => appointment.Doctor.firstName.Contains(appointmentSearchDto.DoctorNameAndSurname) || appointment.Doctor.secondName.Contains(appointmentSearchDto.DoctorNameAndSurname));
+            }
+
+            return appointments;
+        }
+
+        private List<DoctorAppointment> SearchForDate(List<DoctorAppointment> appointments, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (!appointmentSearchDto.Start.Equals("") && !appointmentSearchDto.End.Equals(""))
+            {
+                DateTime startDate = DateTime.ParseExact(appointmentSearchDto.Start, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                DateTime endDate = DateTime.ParseExact(appointmentSearchDto.End, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                appointments = appointments.FindAll(appointment => startDate <= DateTime.ParseExact(appointment.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture) && DateTime.ParseExact(appointment.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture) <= endDate);
+            }
+            else if (appointmentSearchDto.Start.Equals("") && !appointmentSearchDto.End.Equals(""))
+            {
+                DateTime endDate = DateTime.ParseExact(appointmentSearchDto.End, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                appointments = appointments.FindAll(appointment => DateTime.ParseExact(appointment.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture) <= endDate);
+            }
+            else if(!appointmentSearchDto.Start.Equals("") && appointmentSearchDto.End.Equals(""))
+            {
+                DateTime startDate = DateTime.ParseExact(appointmentSearchDto.Start, "dd/MM/yyyy", CultureInfo.InvariantCulture);
+                appointments = appointments.FindAll(appointment => startDate <= DateTime.ParseExact(appointment.Date, "dd/MM/yyyy", CultureInfo.InvariantCulture));
+            }
+            
+            return appointments;
+
+        }
+
+        private List<DoctorAppointment> SearchForAppointmentType(List<DoctorAppointment> appointments, AppointmentReportSearchDto appointmentSearchDto)
+        {
+            if (appointmentSearchDto.AppointmentType.Equals("") || appointmentSearchDto.AppointmentType.Equals("Appointment"))
```

Comment:
Apply a method that checks to see if it is an empty string. Create a method that checks if the appointment type is an appointment and apply it in the if statement.